nemo_text_processing/text_normalization/normalize_with_audio.py

```diff
@@ -148,12 +163,21 @@ def normalize(
 
         if len(normalized_texts) == 0:
             raise ValueError()
-        if punct_post_process:
-            normalized_texts = [post_process_punctuation(t) for t in normalized_texts]
 
+        if punct_post_process:
             # do post-processing based on Moses detokenizer
             if self.processor:
                 normalized_texts = [self.processor.detokenize([t]) for t in normalized_texts]
+                normalized_texts = [
+                    post_process_punct(input=original_text, normalized_text=t) for t in normalized_texts
+                ]
+            else:
+                print("NEMO_NLP collection is not available: skipping punctuation post_processing")
+
+        if self.default_norm:
+            default_norm = self.default_norm.normalize(text=text, punct_post_process=punct_post_process)
+            normalized_texts += [default_norm]
```

Comment:
why do we need this? shouldn't this be covered in normalized_texts by non-deterministic TN?